src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -392,30 +394,46 @@ private SnapshotProject retrieveSnapshotProject(String sql, MapSqlParameterSourc
         String sort,
         String direction,
         String filter,
+        List<UUID> datasetIds,
         List<UUID> accessibleSnapshotIds) {
         logger.debug("retrieve snapshots offset: " + offset + " limit: " + limit + " sort: " + sort +
-            " direction: " + direction + " filter:" + filter);
+            " direction: " + direction + " filter: " + filter + " datasetIds: " + StringUtils.join(datasetIds, ","));
         MapSqlParameterSource params = new MapSqlParameterSource();
         List<String> whereClauses = new ArrayList<>();
-        DaoUtils.addAuthzIdsClause(accessibleSnapshotIds, params, whereClauses);
+        DaoUtils.addAuthzSnapshotIdsClause(accessibleSnapshotIds, params, whereClauses);
         whereClauses.add(" flightid IS NULL"); // exclude snapshots that are exclusively locked
 
         // add the filter to the clause to get the actual items
         DaoUtils.addFilterClause(filter, params, whereClauses);
-        String whereSql = "";
-        if (!whereClauses.isEmpty()) {
-            whereSql = " WHERE " + StringUtils.join(whereClauses, " AND ");
+        String joinSql = "";
+
+        if (datasetIds != null && !datasetIds.isEmpty()) {
```

Comment:
It looks like the only non-test call to this API will never pass `null`. I think it would be better to make the test code conform to the production model rather than adding support for `null`, which should never occur in production.